src/signature/utils.cairo

```diff
@@ -8,12 +8,20 @@ pub fn remove_opcodeseparator(script: @ByteArray) -> @ByteArray {
     let mut parsed_script: ByteArray = "";
     let mut i: usize = 0;
 
+    // TODO: tokenizer/standardize script parsing
     while i < script.len() {
-        let value = script[i];
-        if value != Opcode::OP_CODESEPARATOR {
-            parsed_script.append_byte(value);
+        let opcode = script[i];
+        // TODO: Error handling
+        if opcode == Opcode::OP_CODESEPARATOR {
+            i += 1;
+            continue;
+        }
+        let data_len = Opcode::data_len(i, script).unwrap();
+        let end = i + data_len + 1;
+        while i < end {
+            parsed_script.append_byte(script[i]);
+            i += 1;
         }
-        i += 1;
     };
```

Comment:
@bloomingpeach Looks like there was a bug in this function that caused the transaction to be parsed incorrectly when computing the sighash. After fixing it, you test is passing!